src/features/weather/weatherSlice.js

```diff
@@ -0,0 +1,157 @@
+import axios from "axios"
+import moment from "moment"
+import { createSlice } from "@reduxjs/toolkit"
+
+export const weatherSlice = createSlice({
+    name: "counter",
+    initialState: {
+        currentCity: {}, //Current city displayed
+        previousCity: {}, //Last city searched
+        cities: [], //List of saved cities
+        error: "",
+    },
+    reducers: {
+        //Set current city to display
+        setCurrentCity: (state, action) => {
+            state.previousCity = state.currentCity || action.payload
```

Comment:
Looks like if `state.currentCity` is not defined it should become `action.payload`

But since `state.currentCity` is an object, how this check could be false?

Besides that, what would happen if both `previousCity ` and `currentCity` have the same value?

Reply:
Thanks! I will fix it!
If `previousCity` and `currentCity` have the same value nothing happen, simply if an error occurs then when user clicks back button `currentCity` will be set with `previousCity`.
previousCity is always a valid city, in this way the interface show the last valid city searched by the user